packages/atomic/src/components/atomic-refine-toggle/atomic-refine-toggle.tsx

```diff
@@ -36,11 +37,19 @@ export class AtomicRefineToggle implements InitializableComponent {
 
   public initialize() {
     this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  private loadModal() {
     this.modalRef = document.createElement('atomic-refine-modal');
+    this.modalRef.style.display = 'none';
     this.host.insertAdjacentElement('beforebegin', this.modalRef);
+    this.modalRef.openButton = this.buttonRef;
   }
 
   private enableModal() {
+    if (!this.modalRef) {
+      this.loadModal();
```

Comment:
We won't load the atomic refine modal + atomic modal files until it's needed. I guess most users accessing a search page won't require the modal.

Reply:
Same for the explain why modal, also in this PR